app/controllers/transactions_controller.rb

```diff
@@ -3,7 +3,18 @@ class TransactionsController < ApplicationController
   before_action :set_transaction, only: %i[ show edit update destroy ]
 
   def index
-    @pagy, @transactions = pagy(Current.family.transactions.order(date: :desc), items: 50)
+    search_params = params[:q] || {}
+    if search_params[:date].present?
+      start_date, end_date = calculate_date_range(params[:q][:date])
```

Comment:
We can use the `Period.find_by_name` here:

https://github.com/maybe-finance/maybe/blob/9c9459211fa48abc6f982be0e5215d4cfe2e2b9c/app/models/period.rb#L4-L6

Something like:

```
period = Period.find_by_name(params[:q][:date])
search_params.merge!({date_gteq: period.date_range.begin, date_lteq: period.date_range.end}) if period.date_range
```